### ISSUE TITLE
[IDEA] Translation System with Multiple Languages

### ISSUE THREAD

**Issue by Minionguyjpro** [LumaTeam/Luma3DS#1876]
I would like to see a translation system that allows to have Luma3DS in multiple languages, such as your own one and others.

**Comment by PabloMK7** [comment]
This would be an interesting feature indeed, as people won't need to recompile or modify the source to apply language patches. However it has downsides such as increased binary size because of the extra logic to handle language packs.

**Comment by Minionguyjpro** [comment]
> This would be an interesting feature indeed, as people won't need to recompile or modify the source to apply language patches. However it has downsides such as increased binary size because of the extra logic to handle language packs.

Yes, but I think it would not care for most people though. Otherwise yiu can also make it an optional feature/seperate language pack.

**Comment by Komas19-new** [comment]
or atleast making it easier to modify the texts, for example, in a lang.yml file

**Comment by PoorPocketsMcNewHold** [comment]
> This would be an interesting feature indeed, as people won't need to recompile or modify the source to apply language patches. However it has downsides such as increased binary size because of the extra logic to handle language packs.

You are indeed talking about handling external language packs ? Packing community translation into Luma, while practical would indeed increase too much the binary size, but does loading an external language file placed in the luma folder on the SD perhaps works ?
I'm wondering if the pre-boot luma option menu has access to the SD card content already at this stage to load the translation.

**Comment by Minionguyjpro** [comment]
> > This would be an interesting feature indeed, as people won't need to recompile or modify the source to apply language patches. However it has downsides such as increased binary size because of the extra logic to handle language packs.
> 
> You are indeed talking about handling external language packs ? Packing community translation into Luma, while practical would indeed increase too much the binary size, but does loading an external language file placed in the luma folder on the SD perhaps works ?
> I'm wondering if the pre-boot luma option menu has access to the SD card content already at this stage to load the translation.

Well if Luma at boot can load itself from the SD card then it can also do that I'd say. That's indeed a good solution to the binary sizes problem!
